### ISSUE TITLE
Filtering markers 

### ISSUE THREAD

**Issue by Presidente27** [mapbox/store-locator-android#29]
How can I filter markers based on a feature in  json file? Much like in this example, but it's not for sdk: https://docs.mapbox.com/mapbox-gl-js/example/filter-markers/
Thanks!

**Comment by langsmith** [comment]
Hey @Presidente27 , thanks for looking into Mapbox. Issue tickets in this repo are for problems regarding this app, rather than for Mapbox implementation questions. Please contact our Support team via https://support.mapbox.com/hc/en-us/requests/new?ticket_form_id=360000291231 to get help with this.

**Comment by Presidente27** [comment]
Hi Langston! Actually I was hoping for implementation advice specifically for this app, that's why I posted here. Thanks for the reply anyway.

**Comment by langsmith** [comment]
tldr; the map will have a layer for each category (bar, music, bicycle). 

From your file, you'll eventually have a `FeatureCollection`. Each `Feature` has as many properties in it as you want (https://macwright.org/2015/03/23/geojson-second-bite.html#features and https://tools.ietf.org/html/rfc7946#section-3.2). 

Each `Feature` should have a property field that describes its category. The category will determine which layer the `Feature` will be a part of. 

A checkbox toggle will need to be added to the XML layout of this app's `MapActivity`. Once a category is checked, you'll need to adjust the correct layer's visibility. 

JS version from https://docs.mapbox.com/mapbox-gl-js/example/filter-markers/: ![Screen Shot 2019-06-03 at 11 12 08 AM](https://user-images.githubusercontent.com/4394910/58824555-312a9e80-85f1-11e9-8fe1-327b3ba32791.png)

Java example:
```java
private void toggleLayer() {
    Layer layer = map.getStyle().getLayer("museums");
    if (layer != null) {
      if (VISIBLE.equals(layer.getVisibility().getValue())) {
        layer.setProperties(visibility(NONE));
      } else {
        layer.setProperties(visibility(VISIBLE));
      }
    }
  }
```

**Comment by langsmith** [comment]
@Presidente27 , I spun up a crude example of how to adjust the code to set up toggling per category. Did my work on a branch titled `ls-filtering-toggle` . It's not perfect, but hopefully it's enough to get you started.

https://github.com/mapbox/store-locator-android/pull/new/ls-filtering-toggle

![ezgif com-resize](https://user-images.githubusercontent.com/4394910/58830189-9df86580-85fe-11e9-8df8-f4688e48c23b.gif)

**Comment by Presidente27** [comment]
Hi @langsmith , that's absolutely fantastic, exactly what I needed, definitely enough for me as a starting point. Thanks a lot!

**Comment by langsmith** [comment]
Glad to hear it, @Presidente27 . Happy mapping!
